### PR TITLE
VACMS-4090: Fix required related info field bug.

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
See #4090 

## Testing done
Visual / Behat

## Screenshots
image
image

## QA steps
- As an administrator, go to `node/add/q_a`, and complete all fields required _outside_ of `Enable standalone Resources and support page for this Q&A.` (don't toggle it on).
- Save the form, and confirm no errors are thrown.
- Edit the form, toggle on the `Enable standalone Resources and support page for this Q&A.`, visually verify that a link widget appears for related info, click save, and confirm that error is thrown indicating the related info link field is required.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [x] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by kevwalsh** [comment]
Edit_Q_A_hello___VA_gov_CMS

@ethanteague i may be missing something, but i don't see any difference here. 

1. Related info is still not required
2. No empty link teaser form is displayed upon load. 

https://pr4189-0bmtpopyxrr6z4bnuj5o3v6wf7zempym.ci.cms.va.gov/node/15848/edit

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 1]
```diff
@@ -795,14 +795,17 @@ function _va_gov_backend_audience_topics_validation(array $form, FormStateInterf
...
+ * Custom validation callback for the primary category and related info fields.
  *
  * @param array $form
  *   The exposed widget form array.
  * @param \Drupal\Core\Form\FormStateInterface $form_state
  *   The form state.
  */
 function _va_gov_backend_standalone_page_validation(array &$form, FormStateInterface &$form_state) {
+  if ($form_state->getValue('field_standalone_page')['value'] === 0) {
+    $form_state->unsetValue('field_related_information');
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:807]: Could benefit from a comment here describing the why.

**Comment by ethanteague** [comment]
> Edit_Q_A_hello___VA_gov_CMS
> 
> @ethanteague i may be missing something, but i don't see any difference here.
> 
> 1. Related info is still not required
> 2. No empty link teaser form is displayed upon load.
> 
> https://pr4189-0bmtpopyxrr6z4bnuj5o3v6wf7zempym.ci.cms.va.gov/node/15848/edit

Ready for another pass when you have a sec, @kevwalsh

**Comment by swirtSJW** [comment]
QA did not pass
image


image

**Comment by ethanteague** [comment]
> QA did not pass
> image
> 
> image

Should be fixed now, @swirtSJW

**Comment by swirtSJW** [comment]
@ethanteague I am still getting that same error on the pr environment.  Even cleared my caches.
image

**Review by swirtSJW — CHANGES_REQUESTED** [review]
QA not passing  see previous comment.

**Comment by ethanteague** [comment]
> QA not passing see previous comment.

Dangit.......sorry about this, @swirtSJW ! Lemme take a look.

**Review by swirtSJW — APPROVED** [review]
Tests out fine.   Nice work.

**Comment by swirtSJW** [comment]
This is good to go.  Just waiting on tests to pass.  
When they pass it can be merged.

**Pull request merged**